Code/Framework/AzToolsFramework/AzToolsFramework/Slice/SliceDependencyBrowserComponent.cpp

```diff
@@ -222,33 +222,53 @@ namespace AzToolsFramework
         }
 
         bool succeeded = m_databaseConnection->QueryDependsOnSourceBySourceDependency(
-            sourceUuid, "%.slice",
+            sourceUuid,
             AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::DEP_SourceOrJob,
             [&](AssetDatabase::SourceFileDependencyEntry& entry)
-        {
-            found = true;
-            dependencies.push_back(AZStd::move(entry.m_dependsOnSource));
-            return true;
-        });
+            {
+                AZStd::string dependencyName = entry.m_dependsOnSource.GetPath();
+
+                if (entry.m_dependsOnSource.IsUuid())
+                {
+                    AzToolsFramework::AssetDatabase::SourceDatabaseEntry sourceEntry;
+                    m_databaseConnection->QuerySourceBySourceGuid(
+                        entry.m_dependsOnSource.GetUuid(),
+                        [&dependencyName](AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
+                        {
+                            dependencyName = entry.m_sourceName;
+                            return false;
+                        });
+                }
+
+                if (!dependencyName.ends_with(".slice"))
+                {
+                    // Filter out non-slice files
```

Comment:
could this comment be expanded to explain why the *.slice names need to be filtered out?